### PR TITLE
Add info and execution to Insomnia SDK with collection runner[INS-4320]

### PR DESCRIPTION
**Pull request by cwangsmv**
Expose more features in SDK with collection runner: 

Task List:
- [x] Add info and execution to SDK property
- [x] Integrate with request and collection runner for skipRequest and setNextRequest method
- [x] Add unit test

### CODE REVIEW

File: packages/insomnia/src/ui/routes/runner.tsx [hunk 1]
```diff
@@ -707,8 +707,18 @@ export const runCollectionAction: ActionFunction = async ({ request, params }) =
   });
 
   for (let i = 0; i < iterations; i++) {
+    // nextRequestIdOrName is used to manual set next request in iteration from pre-request script
+    let nextRequestIdOrName = '';
     for (let j = 0; j < requests.length; j++) {
-      const targetRequest = requests[j];
+      const targetRequest = requests[j] as { name: string; id: string; url: string };
+      if (typeof nextRequestIdOrName === 'string' && nextRequestIdOrName !== '') {
+        if (targetRequest.id === nextRequestIdOrName || targetRequest.name.trim() === nextRequestIdOrName.trim()) {
```
Comment (ihexxa) [inline packages/insomnia/src/ui/routes/runner.tsx:714]: This condition seems satisfied in the first match while we noticed that it tries to find the last match.

File: packages/insomnia/src/network/network.ts [hunk 2]
```diff
@@ -157,10 +166,7 @@ export const tryToExecutePreRequestScript = async ({
     activeGlobalEnvironment = await models.environment.getById(workspaceMeta.activeGlobalEnvironmentId) || undefined;
   }
 
-  const requestGroups = await db.withAncestors<Request | RequestGroup>(request, [
-    models.requestGroup.type,
-  ]) as (Request | RequestGroup)[];
-
+  const requestGroups = ancestors.filter(doc => isRequest(doc) || isRequestGroup(doc)) as RequestGroup[];
```
Comment (ihexxa) [inline packages/insomnia/src/network/network.ts:170]: If this implementation is incorrect, would you mind also modify the same part in the after-response function.

File: packages/insomnia/src/network/network.ts [hunk 3]
```diff
@@ -307,12 +321,20 @@ export const tryToExecuteScript = async (context: RequestAndContextAndOptionalRe
         clientCertificates,
         settings,
         cookieJar,
+        requestInfo: {
+          eventName: 'prerequest',
```
Comment (ihexxa) [inline packages/insomnia/src/network/network.ts]: Besides `prerequest`, it could also be `test` when it is called by `trytoExecuteAfterResponseScript`

**Review by ihexxa — COMMENTED** [review]
Had a quick review and will dive deeper later.

File: packages/insomnia/src/network/network.ts (hunk 3 above)
Comment (cwangsmv) [reply packages/insomnia/src/network/network.ts]: Good catch, I will add the missing test eventName

File: packages/insomnia/src/network/network.ts (hunk 2 above)
Comment (cwangsmv) [reply packages/insomnia/src/network/network.ts:170]: Yes, I have also change the same logic in https://github.com/Kong/insomnia/pull/7859/files#diff-cc2479e11623891c8c7a329087ad50c1f140db4b1e753ea94af01e59af95484dR451.

This change is an enhancement to remove duplicated db call. For both tryToExecuteAfterResponseScript and tryToExecutePreRequestScript function, they will accept requestData variable from fetchRequestData. And I have changed fetchRequestData to fetch all ancestors for the request, so in tryToExecuteAfterResponseScript and  tryToExecutePreRequestScript, they can just use the ancestors property of requestData` variable and reduce db call.

File: packages/insomnia/src/ui/routes/runner.tsx (hunk 1 above)
Comment (cwangsmv) [reply packages/insomnia/src/ui/routes/runner.tsx:714]: Fixed

**Review by ihexxa — APPROVED** [review]

**Pull request merged**